experimental/SetPartitions/src/SetPartition.jl

```diff
@@ -0,0 +1,293 @@
+automatically converted into normal form as returned by [`normal_form`](@ref).
+
+# Examples
+```jldoctest
+julia> SetPartition([2, 4], [4, 99])
+SetPartition([1, 2], [2, 3])
+```
+"""
+struct SetPartition <: AbstractPartition
+    upper_points::Vector{Int}
+    lower_points::Vector{Int}
+
+    function SetPartition(upper_points, lower_points)
+        (new_upper, new_lower) = normal_form(convert(Vector{Int}, upper_points), 
+                                             convert(Vector{Int}, lower_points))
+        return new(new_upper, new_lower)
+    end
+end
+
+
+function hash(p::SetPartition, h::UInt)
+    return hash(p.upper_points, hash(p.lower_points, h))
+end
+
+function ==(p::SetPartition, q::SetPartition)
+    return p.lower_points == q.lower_points && p.upper_points == q.upper_points
+end
+
+function copy(p::SetPartition)
+    return SetPartition(copy(p.upper_points), copy(p.lower_points))
+end
+
+"""
+    tensor_product(p::SetPartition, q::SetPartition)
+
+Return tensor product of `p` and `q`.
+
+The tensor product of two partitions is given by their vertical concatenation.
+See also Section 4.1.1 in [Gro20](@cite).
+
+# Examples
+```jldoctest
+julia> tensor_product(SetPartition([1, 2], [2, 1]), SetPartition([1, 1], [1]))
+SetPartition([1, 2, 3, 3], [2, 1, 3])
+```
+"""
+function tensor_product(p::SetPartition, q::SetPartition)
+    
+    q_new = new_point_values(p.upper_points, p.lower_points, 
+                             q.upper_points, q.lower_points)
+    return SetPartition(vcat(p.upper_points, q_new[1]), vcat(p.lower_points, q_new[2]))
+end
+
+"""
+    involution(p::SetPartition)
+
+Return involution of `p`.
+
+The involution of a partition is obtained by swapping the upper and lower points.
+See also Section 4.1.1 in [Gro20](@cite).
+
+# Examples
+```jldoctest
+julia> involution(SetPartition([1, 2, 3], [2, 1]))
+SetPartition([1, 2], [2, 1, 3])
+```
+"""
+function involution(p::SetPartition)
+    return SetPartition(p.lower_points, p.upper_points)
+end
+
+"""
+    vertical_reflection(p::SetPartition)
+
+Return vertical reflection of `p`.
+
+The vertical reflection of a partition is obtained by reversing the order of 
+the upper and lower points. See also Section 4.1.2 in [Gro20](@cite).
+
+# Examples
+```jldoctest
+julia> vertical_reflection(SetPartition([1, 2, 3], [2, 1]))
+SetPartition([1, 2, 3], [3, 2])
+```
+"""
+function vertical_reflection(p::SetPartition)
+    return SetPartition(reverse(p.upper_points), reverse(p.lower_points))
+end
+
+"""
+    rotation(p::SetPartition)
+
+Return the rotation of `p` in the direction given by `lr` and `tb`. 
+
+Rotating a partition moves the left- or right-most point of the upper points 
+to the lower points or vice verca. See also Section 4.1.2 in [Gro20](@cite).
+
+# Arguments
+- `p`: input partition
+- `lr`: rotating at the left (true) or at the right (false)
+- `tb`: rotating from top to bottom (true) or from bottom to top (false)
```

Comment:
These are not present in the signature in Line 100, and are still not explained properly. Having two booleans as positional arguments seems the seed of future confusion for me, as one needs to remember their order. Maybe it would be better to have them as keyword arguments instead.

Reply:
Would it be better to split it into four functions like `rotate_top_left`, `rotate_top_right`, ... and then document it separately?